tests/unit/neptune/new/api/test_searching_entries.py

```diff
@@ -68,39 +74,322 @@ def test__to_leaderboard_entry():
+            LeaderboardEntry(
+                id="foo",
+                attributes=[
+                    AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "h"})
+                ],
+            ),
+            LeaderboardEntry(
+                id="foo",
+                attributes=[
+                    AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "j"})
+                ],
+            ),
+        ],
+        None,
+    ]
 
     # when
-    result = list(iter_over_pages(iter_once=iter_once, step=3))
+    result = list(iter_over_pages(step_size=3))
 
     # then
-    assert result == [1, 2, 3, 4, 5, 6, 7, 8, 9]
+    assert result == [
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "a"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "b"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "c"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "d"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "e"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "f"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "g"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "h"})],
+        ),
+        LeaderboardEntry(
+            id="foo",
+            attributes=[AttributeWithProperties(path="sys/id", type=AttributeType.STRING, properties={"value": "j"})],
+        ),
+    ]
+    assert get_single_page.mock_calls == [
+        call(limit=3, offset=0, sort_by="sys/id", searching_after=None),
+        call(limit=3, offset=3, sort_by="sys/id", searching_after=None),
+        call(limit=3, offset=6, sort_by="sys/id", searching_after=None),
+        call(limit=3, offset=9, sort_by="sys/id", searching_after=None),
+    ]
 
 
-def test__iter_over_pages__empty():
+@patch("neptune.api.searching_entries.get_single_page")
+def test__iter_over_pages__multiple_search_after(get_single_page):
     # given
-    def iter_once(limit: int, offset: int):
-        return []
+    get_single_page.side_effect = [
```

Comment:
As it is repeated in a few tests (almost the same way) maybe it would be worth it to move it to a fixture? It's a little hard to navigate those tests with those long lists

Reply:
Done